plugins/inputs/logfile/fileconfig.go

```diff
@@ -85,6 +96,25 @@ type FileConfig struct {
 	BlacklistRegexP *regexp.Regexp
 	//Decoder object
 	Enc encoding.Encoding
+	sampleCount int
+}
+
+type LogFilter struct {
+	Type string `toml:"type"`
+	Expression string `toml:"expression"`
+	ExpressionP *regexp.Regexp
```

Comment:
If there is no intent to use `ExpressionP` outside the `logfile` package, then you can rename it starting with a lowercase.

It looks like the initialization and use of the `Regexp` struct is confined to this file.

Reply:
Good call out. I can change that. Was really just following what was already there for the timestamp expression